internal/docstore/memdocstore/mem.go

```diff
@@ -160,6 +176,33 @@ func (c *collection) update(doc map[string]interface{}, mods []driver.Mod) error
 			_ = setAtFieldPath(doc, m.FieldPath, m.Value)
 		}
 	}
+	if len(mods) > 0 {
+		c.changeRevision(doc)
```

Comment:
What's the relationship between `c.nextRevision` and `c.docs[key][docstore.RevisionField]`? It seems the collection's revision is changed along with the the docs' revision when doing replace and put, should the docs' revision also be changed here during update?

Reply:
`c.nextRevision` is just a collection-wide counter, to make sure every write gets a unique value.

My intent is that in all three cases (replace, put and update), the stored document's revision (value of its RevisionField) is changed to a new value. Is that not happening somewhere?

Reply:
Ok, got it. No, I think I must have missed something, this looks correct.